pkg/azuredisk/azure_controller_common.go

```diff
@@ -250,14 +247,22 @@ func (c *controllerCommon) AttachDisk(ctx context.Context, diskName, diskURI str
 		}
 	}()
 
-	var future *azure.Future
-	future, err = vmset.AttachDisk(ctx, nodeName, diskMap)
+	err = vmset.AttachDisk(ctx, nodeName, diskMap)
 	if err != nil {
-		return -1, err
-	}
-	// err will be handled by waitForUpdateResult below
-	if err = c.waitForUpdateResult(ctx, vmset, nodeName, future, err); err != nil {
-		return -1, err
+		if IsOperationPreempted(err) {
+			if derr := kwait.ExponentialBackoffWithContext(ctx, updateVMBackoff, func(ctx context.Context) (bool, error) {
```

Comment:
not necessary to ExponentialBackoffWithContext since the CSI driver would retry if failed, let's just keep the original logic, thanks.

Reply:
removed.